distributed/scheduler.py

```diff
@@ -1997,24 +1978,22 @@ def transition_released_waiting(self, key, stimulus_id):
             assert not any([dts.state == "forgotten" for dts in ts.dependencies])
 
         if ts.has_lost_dependencies:
-            recommendations[key] = "forgotten"
-            return recommendations, client_msgs, worker_msgs
+            return {key: "forgotten"}, {}, {}
 
         ts.state = "waiting"
 
-        dts: TaskState
         for dts in ts.dependencies:
             if dts.exception_blame:
                 ts.exception_blame = dts.exception_blame
-                recommendations[key] = "erred"
-                return recommendations, client_msgs, worker_msgs
+                return {key: "erred"}, {}, {}
```

Comment:
This codecov failure is worrying. Not a regression in this PR though. CC @fjetter

Reply:
I think this can be safely deleted. I don't see how we'd transition a task to waiting of any of its dependencies is erred